platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/resolver/DefaultResolutionOutputs.java

```diff
@@ -171,7 +172,8 @@ public ArtifactCollectionInternal getArtifacts() {
                 getFiles(),
                 lenient,
                 resolutionAccess.getHost(),
-                calculatedValueContainerFactory
+                calculatedValueContainerFactory,
+                new AttributeDesugaring(attributesFactory)
```

Comment:
AttibuteDesugaring contains a [cache](https://github.com/gradle/gradle/blob/35bd9ab866a9721114fd3dd85e7237c4b8e42153/platforms/software/dependency-management/src/main/java/org/gradle/api/internal/attributes/AttributeDesugaring.java#L34) of desugared containers. We should therefore use an injected `AttributeDesugaring` instance instead of creating a new one every time. 

Unfortunately, this does mean we need a bit more wiring, but I think we need to wire the common instance of `AttributeDesugaring` here